apis/v1alpha1/httproute_types.go

```diff
@@ -377,70 +396,116 @@ type HTTPRequestHeaderFilter struct {
+	// +kubebuilder:validation:MaxLength=253
+	ServiceName *string `json:"serviceName,omitempty"`
+
+	// BackendRef is a local object reference to mirror matched requests to. If
+	// both BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this Route
+	// to true.
+	//
+	// Support: Extended
 	//
-	TargetRef ForwardToTargetObjectReference `json:"targetRef" protobuf:"bytes,1,opt,name=targetRef"`
+	// +optional
+	BackendRef LocalObjectReference `json:"backendRef,omitempty"`
 
-	// TargetPort specifies the destination port number to use for the TargetRef.
-	// If unspecified and TargetRef is a Service object consisting of a single
-	// port definition, that port will be used. If unspecified and TargetRef is
-	// a Service object consisting of multiple port definitions, an error is
-	// surfaced in status.
+	// Port specifies the destination port number to use for the backend. If
+	// unspecified and a Service object consisting of a single port definition
+	// is the backend, that port will be used. If unspecified and the backend is
+	// a Service object consisting of multiple port definitions, controllers
+	// must set the "InvalidRoutes" status condition on any Gateway that
+	// includes this Route to true.
 	//
 	// Support: Core
 	//
 	// +optional
-	TargetPort *TargetPort `json:"targetPort,omitempty" protobuf:"bytes,2,opt,name=targetPort"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	Port *int32 `json:"port,omitempty"`
 }
 
-// HTTPForwardingTarget is the target to send the request to for a given a match.
-type HTTPForwardingTarget struct {
-	// To references referenced object(s) where the request should be sent. The
-	// resource may be "services" (omit or use the empty string for the
-	// group), or an implementation may support other resources (for
-	// example, resource "myroutetargets" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "services".  If the referent
-	// cannot be found, the "InvalidRoutes" status condition on any Gateway
-	// that includes the HTTPRoute will be true.
+// HTTPRouteForwardTo defines how a HTTPRoute should forward a request.
+type HTTPRouteForwardTo struct {
+	// ServiceName refers to the name of the Service to forward matched requests
+	// to. When specified, this takes the place of BackendRef. If both
+	// BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this
+	// Route to true.
 	//
-	// Support: core
+	// Support: Core
 	//
-	// +kubebuilder:validation:MinItems=1
-	To []HTTPForwardToTarget `json:"to"`
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
```

Comment:
Same as https://github.com/kubernetes-sigs/service-apis/pull/336/files#r492313616

Reply:
Why would we add a minLength validation of 1 on an optional field?